baremetal_network_provisioning/ml2/extensions/switch.py

```diff
@@ -12,29 +12,225 @@
 #    under the License.from oslo.config import cfg
+
+from simplejson import scanner as json_scanner
+
 from neutron.api import extensions
+from neutron.api.v2 import attributes
+from neutron import context
 from neutron import wsgi
 
+from baremetal_network_provisioning.common import constants as const
+from baremetal_network_provisioning.common import exceptions as exc
+from baremetal_network_provisioning.db import bm_nw_provision_db as db
+from baremetal_network_provisioning.drivers import discovery_driver
+
+from oslo_log import log as logging
+
+LOG = logging.getLogger(__name__)
+
 
-EXTENDED_ATTRIBUTES_2_0 = {
-    "switches": {
-    }
+RESOURCE_ATTRIBUTE_MAP = {
+    'switches': {
+        'id': {'allow_post': False, 'allow_put': False,
+               'is_visible': True},
+        'ip_address': {'allow_post': True, 'allow_put': False,
+                       'validate': {'type:string': None},
+                       'is_visible': True, 'default': ''},
+        'mac_address': {'allow_post': True, 'allow_put': False,
+                        'validate': {'type:string': None},
+                        'is_visible': True, 'default': ''},
+        'access-parameters': {'allow_post': True, 'allow_put': True,
+                              'validate': {'type:access_dict': None},
+                              'is_visible': True},
+        'access-protocol': {'allow_post': True, 'allow_put': True,
+                            'validate': {'type:string': None},
+                            'is_visible': True},
+        'vendor': {'allow_post': True, 'allow_put': False,
+                   'validate': {'type:string': None},
+                   'is_vidible': True}
+    },
 }
 
 
+def validator_func(data, valid_values=None):
+    """Validate the access parameters."""
+    if not data:
+        # Access parameters must be provided.
+        msg = _("Cannot create a switch from the given input.")
+        return msg
+    if type(data) is not dict:
+        msg = _("Given details is not in the form of a dictionary.")
+        return msg
+
+attributes.validators['type:access_dict'] = validator_func
+
+
 class BNPSwitchController(wsgi.Controller):
 
+    @property
+    def _dbcontext(self):
+        return context.get_admin_context()
+
     def index(self, request):
-        pass
+        switches = db.get_all_bnp_phys_switches(self._dbcontext)
+        switches_dict = {
+            'switches': [switch.__dict__ for switch in switches]}
+        return switches_dict
 
     def show(self, request, id):
-        pass
+        switch = db.get_bnp_phys_switch(self._dbcontext, id)
+        if not switch:
+            raise exc.NotFound(
+                resource="switch %s" % (id))
+        switch_dict = switch.__dict__
+        switch_dict['ports'] = []
+        # Get a list of Ironic ports
+        return switch_dict
 
     def delete(self, request, id):
-        pass
+        switch = db.get_bnp_phys_switch(self._dbcontext, id)
```

Comment:
Yes because DB does not throw an error if a wrong ID is passed. Then user will not come to know if he has deleted the right switch